pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -212,15 +246,18 @@ def _find_names(self, lr_schedulers: List, add_lr_sch_names: bool = True) -> Lis
 
             # Multiple param groups for the same scheduler
             param_groups = sch.optimizer.param_groups
+            duplicates = self._duplicate_param_group_names(param_groups)
+            if duplicates:
+                raise MisconfigurationException(
+                    'A single `Optimizer` cannot have multiple parameter groups with identical '
+                    f'`name` values. {name} has duplicated parameter group names {duplicates}'
+                )
 
             name = self._add_prefix(name, optimizer_cls, seen_optimizer_types)
 
-            if len(param_groups) != 1:
-                for i in range(len(param_groups)):
-                    temp = f'{name}/pg{i + 1}'
-                    names.append(temp)
-            else:
-                names.append(name)
+            for i in range(len(param_groups)):
+                name_and_suffix = self._add_suffix(name, param_groups, i)
+                names.append(name_and_suffix)
```

Comment:
Could be one-line :D

```suggestion
            names.extend(self._add_suffix(name, param_groups, i) for i in range(len(param_groups)))
```

Reply:
I'm on the fence here. I think I prefer the readability++ of the intermediate variable name, but I don't feel strongly enough about it to oppose those who would prefer this suggestion.

Reply:
Hey @AffineParameter, we would prefer if you could use list comprehension there :)